src/main/java/org/duraspace/bagit/BagConfig.java

```diff
@@ -63,35 +64,26 @@
 
     public static final String ACCESS_KEY = "Access";
 
-    private Map<String, Map<String, String>> map;
+    private final Map<String, Map<String, String>> map;
 
     /**
      * Default constructor
      *
-     * @param bagConfigFile a bagit config yaml file (see src/test/resources/bagit-config.yml)
+     * @param bagConfigReader a reader for a bagit config yaml file (see src/test/resources/bagit-config.yml)
      */
     @SuppressWarnings("unchecked")
-    public BagConfig(final File bagConfigFile) {
-        final String bagConfigFilePath = bagConfigFile.getAbsolutePath();
-
-        YamlReader reader = null;
+    public BagConfig(final Reader bagConfigReader) {
+        YamlReader yaml = null;
         try {
-            reader = new YamlReader(Files.newBufferedReader(bagConfigFile.toPath(), UTF_8));
-            map = (Map<String, Map<String, String>>) reader.read();
-            if (getBagInfo() == null) {
-                throw new RuntimeException("The " + BAG_INFO_KEY + " key is not present in the " + bagConfigFilePath);
-            }
-
-
-        } catch (FileNotFoundException e) {
-            throw new RuntimeException("The specified bag config file does not exist: " + bagConfigFile
-                    .getAbsolutePath());
-        } catch (Exception e) {
-            throw new RuntimeException("The specified bag config file could not be parsed: " + e.getMessage(), e);
+            yaml = new YamlReader(bagConfigReader);
+            map = (Map<String, Map<String, String>>) yaml.read();
+        } catch (YamlException e) {
+            logger.error("Unable to parse yaml");
```

Comment:
Maybe add the `e` exception to the log in order to get the full details?

Reply:
I was considering that as well, added it in the latest commit